app/models/product.rb

```diff
@@ -0,0 +1,16 @@
+class Product < ApplicationRecord
+  belongs_to :user
+  belongs_to :category
+  has_one :order_item, dependent: :destroy, optional: true
```

Comment:
sao lại là has_one cậu
1 product có nhìu order item chứ, 1 order item thì thuộc về 1 product

Reply:
```suggestion
  has_many :order_items, dependent: :destroy, optional: true
```

ngoài ra còn 1 lệnh nữa là
```
has_many :orders, through: :order_items .....
```